crates/zk-por-cli/src/merkle_proof.rs

```diff
@@ -0,0 +1,57 @@
+use std::{fs::File, io::Write, str::FromStr};
+
+use serde_json::json;
+use zk_por_core::{
+    config::ProverConfig,
+    database::{DataBase, DbOption},
+    error::PoRError,
+    global::GlobalConfig,
+    merkle_proof::MerkleProof,
+    parser::{AccountParser, FileAccountReader, FileManager, FilesCfg},
+};
+
+use crate::constant::RECURSION_BRANCHOUT_NUM;
+
+pub fn get_merkle_proof(
+    user_id: String,
+    cfg: ProverConfig,
+    output_path: String,
+) -> Result<(), PoRError> {
+    let database = DataBase::new(DbOption {
+        user_map_dir: cfg.db.level_db_user_path.to_string(),
+        gmst_dir: cfg.db.level_db_gmst_path.to_string(),
+    });
+
+    let batch_size = cfg.prover.batch_size as usize;
+    let token_num = cfg.prover.num_of_tokens as usize;
+
+    // the path to dump the final generated proof
+    let file_manager = FileManager {};
+    let account_parser = FileAccountReader::new(
+        FilesCfg {
+            dir: std::path::PathBuf::from_str(&cfg.prover.user_data_path).unwrap(),
+            batch_size: cfg.prover.batch_size,
+            num_of_tokens: cfg.prover.num_of_tokens,
+        },
+        &file_manager,
+    );
+    account_parser.log_state();
+    // let mut account_parser: Box<dyn AccountParser> = Box::new(parser);
+
+    let batch_num = account_parser.total_num_of_users().div_ceil(batch_size);
```

Comment:
the account parser will read the first and last files; if that's the case, we will have a master db to save relevant meta info to DB. can u help refactor that part

Reply:
Hmm maybe we can discuss this